spec/hyacinth/utils/clean_spec.rb

```diff
@@ -0,0 +1,143 @@
+          }
+        ],
+        'an_array_field' => ['some_value  ', '     ']
+      }
+    end
+
+    let(:expected_hash_without_whitespace) do
+      {
+        'alternate_title' => [
+          {
+            'title_non_sort_portion' => 'No Extra Spaces',
+            'title_sort_portion' => 'Catcher in the Rye'
+          }
+        ],
+        'name' => [
+          {
+            'name_value' => 'This has space',
+            'name_role' => [
+              {
+                'name_role_value' => 'This value has spaaaace'
+              },
+              {
+                'name_role_value' => 'Unchanged role without extra space'
+              }
+            ]
+          }
+        ],
+        'some_field_group' => [
+          {
+            'some_field' => 'Great value here',
+            'controlled_field' => {
+              'controlled_field_uri' => 'http://id.library.columbia.edu/with/leading/space',
+              'controlled_field_value' => 'Value with trailing space'
+            }
+          }
+        ],
+        'an_array_field' => ['some_value', '']
+      }
+    end
+
+    it 'cleans data structure as expected' do
+      expect(described_class.trim_whitespace!(hash_with_whitespace)).to eq(expected_hash_without_whitespace)
+      expect(hash_with_whitespace).to eq(expected_hash_without_whitespace)
+    end
+  end
+
+  describe '.remove_blank_fields!' do
+    let(:hash_with_blank_fields) do
+      {
+        'alternate_title' => [
+          {
+            'title_non_sort_portion' => '',
+            'title_sort_portion' => 'Catcher in the Rye'
+          }
+        ],
+        'name' => [
+          {
+            'name_value' => '',
+            'name_role' => [
+              {
+                'name_role_value' => ''
+              }
+            ]
+          }
+        ],
+        'controlled_field_group' => [
+          {
+            'controlled_field' => {
+              'controlled_field_uri' => '',
+              'controlled_field_value' => ''
+            }
+          }
+        ],
+        'collection' => [
+          {
+            'collection_authorized_term_uri' => 'http://example.com/a/b/c'
+          }
+        ],
+        'note' => [
+          {
+            'note_value' => '                         ', # A bunch of spaces
+            'note_type' => ''
+          }
+        ],
+        'an_array_field' => ['', nil, 'some_value', ['', nil]]
+      }
+    end
+
+    let(:expected_hash_without_blank_fields) do
+      {
+        'alternate_title' => [
+          {
+            'title_sort_portion' => 'Catcher in the Rye'
+          }
+        ],
+        'collection' => [
+          {
+            'collection_authorized_term_uri' => 'http://example.com/a/b/c'
+          }
+        ],
+        'an_array_field' => ['some_value']
+      }
+    end
+
+    it 'cleans data as expected' do
+      expect(described_class.remove_blank_fields!(hash_with_blank_fields)).to eq(expected_hash_without_blank_fields)
```

Comment:
```suggestion
      expect(described_class.remove_blank_fields!(hash_with_blank_fields)).to eq(expected_hash_without_blank_fields)
      expect(hash_with_blank_fields).to eq(expected_hash_without_blank_fields)
```
Just to prove that the original has been modified.